src/common.rs

```diff
@@ -11,27 +11,82 @@ pub trait AsU32 {
 cfg_if::cfg_if! {
     if #[cfg(any(windows, target_os = "unknown", target_arch = "wasm32"))] {
         /// Process id.
-        pub type Pid = usize;
+        #[derive(Debug, Clone, Copy, Eq, Hash)]
+        pub struct Pid(usize);
 
-        impl AsU32 for Pid {
-            fn as_u32(&self) -> u32 {
-                *self as u32
+        impl From<u32> for Pid {
+            fn from(u32v: u32) -> Pid {
+                Pid(u32v as usize)
             }
         }
+
+        impl From<i32> for Pid {
+            fn from(i32v: i32) -> Pid {
+                Pid(i32v as usize)
+            }
+        }
+
     } else {
         use libc::pid_t;
 
         /// Process id.
-        pub type Pid = pid_t;
+        #[derive(Debug, Clone, Copy, Eq, Hash)]
+        pub struct Pid(pid_t);
+
+        impl From<u32> for Pid {
+            fn from(u32v: u32) -> Pid {
+                Pid(u32v as pid_t)
+            }
+        }
 
-        impl AsU32 for Pid {
-            fn as_u32(&self) -> u32 {
-                *self as u32
+        impl From<i32> for Pid {
+            fn from(i32v: i32) -> Pid {
+                Pid(i32v as pid_t)
             }
         }
     }
 }
 
+impl PartialEq for Pid {
+    fn eq(&self, other: &Self) -> bool {
+        self.0 == other.0
+    }
+}
+
+impl From<Pid> for u32 {
```

Comment:
I don't think it's needed, the `Into` trait is implemented automatically (take a look at the explanation [here](https://doc.rust-lang.org/nightly/std/convert/trait.From.html#generic-implementations)).